packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -87,7 +87,8 @@ function getEngineApiKey(engine: Config['engine']): string | undefined {
 
 function getEngineServiceId(engine: Config['engine']): string | undefined {
   const engineApiKey = getEngineApiKey(engine);
-  if (engineApiKey) {
+
+  if (engine !== false && engineApiKey) {
```

Comment:
Seems like this logic should be in getEngineApiKey?

Reply:
indeed, fixed